sixtyfps_runtime/rendering_backends/gl/lib.rs

```diff
@@ -677,120 +608,124 @@ impl ItemRenderer for GLItemRenderer {
+                    // visual end of glyph before the cursor, because due to for example ligatures
+                    // (or generally glyph substitution) there may not be a dedicated glyph.
+                    for glyph in &metrics.glyphs {
+                        if glyph.byte_index == min_select.saturating_sub(start) {
+                            selection_start_x = glyph.x - glyph.bearing_x;
+                        }
+                        if glyph.byte_index == max_select - start
+                            || glyph.byte_index >= to_draw.len()
+                        {
+                            after_selection_x = glyph.x - glyph.bearing_x;
+                            break;
+                        }
+                        selection_end_x = glyph.x + glyph.advance_x;
+                    }
 
-        // This way of drawing selected text isn't quite 100% correct. Due to femtovg only being able to
-        // have a simple rectangular selection - due to the use of the scissor clip - the selected text is
-        // drawn *over* the unselected text. If the selection background color is transparent, then that means
-        // that glyphs are blended twice, which may lead to artifacts.
-        // It would be better to draw the selected text and non-selected text without overlap.
-        if text_input.has_selection() {
-            let (anchor_pos, cursor_pos) = text_input.selection_anchor_and_cursor();
-            let mut selection_start_x = 0.;
-            let mut selection_end_x = 0.;
-            // Determine the first and last (inclusive) glyph of the selection. The anchor
-            // will always be at the start of a grapheme boundary, so there's at ShapedGlyph
-            // that has a matching byte index. For the selection end we have to look for the
-            // visual end of glyph before the cursor, because due to for example ligatures
-            // (or generally glyph substitution) there may not be a dedicated glyph.
-            for glyph in &metrics.glyphs {
-                if glyph.byte_index == anchor_pos {
-                    selection_start_x = glyph.x;
-                }
-                if glyph.byte_index == cursor_pos {
-                    break;
+                    let selection_rect = Rect::new(
+                        pos + euclid::vec2(selection_start_x, 0.),
+                        Size::new(selection_end_x - selection_start_x, font_height),
+                    );
+                    canvas.fill_path(
+                        &mut rect_to_path(selection_rect),
+                        femtovg::Paint::color(to_femtovg_color(
+                            &text_input.selection_background_color(),
+                        )),
+                    );
+                    let mut selected_paint = paint;
+                    selected_paint
+                        .set_color(to_femtovg_color(&text_input.selection_foreground_color()));
+                    canvas
```

Comment:
Like the old code had a comment about the paint order, I think this code deserves a comment that it needs fixing to work with text selection in ligatures. Or maybe a reminder that this all should go into femtovg.